backend/src/main/java/com/javahelp/backend/query/VanillaConstraint.java

```diff
@@ -0,0 +1,10 @@
+package com.javahelp.backend.query;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class VanillaConstraint extends Constraint{
```

Comment:
What is a VanillaConstraint?

Reply:
It is just a very simply constraint that finds matching attribute values through equivalence relation.